crates/oxc_linter/src/rules/unicorn/consistent_function_scoping.rs

```diff
@@ -161,12 +161,20 @@ impl Rule for ConsistentFunctionScoping {
     fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
         let (function_declaration_symbol_id, function_body, reporter_span) = match node.kind() {
             AstKind::Function(function) => {
-                if let Some(AstKind::AssignmentExpression(_)) = ctx.nodes().parent_kind(node.id()) {
+                if function.is_typescript_syntax() {
                     return;
                 }
 
-                if function.is_typescript_syntax() {
-                    return;
+                if let Some(parent_scope_id) =
+                    ctx.scopes().get_parent_id(function.scope_id.get().unwrap())
```

Comment:
Don't unwrap here, we just had a bug where this exact logic caused panics in the linter.

Reply:
Saw this in some other place, thought it would be fine.

Reply:
Existing bad practices should be fixed, not followed